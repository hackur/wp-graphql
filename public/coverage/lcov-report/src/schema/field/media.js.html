<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/schema/field/media.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/schema/field</a> media.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/14</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { Grap<span class="cstat-no" title="statement not covered" >hQLString } from 'graphql';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span>
import MEDIA_TYPE<span class="cstat-no" title="statement not covered" > from 'enum/MediaType';<span class="cstat-no" title="statement not covered" ></span></span>
import Description <span class="cstat-no" title="statement not covered" >from 'type/Description';<span class="cstat-no" title="statement not covered" ></span></span>
import Caption <span class="cstat-no" title="statement not covered" >from 'type/Caption';<span class="cstat-no" title="statement not covered" ></span></span>
import MediaT<span class="cstat-no" title="statement not covered" >ype from 'type/Media';<span class="cstat-no" title="statement not covered" ></span></span>
import Media f<span class="cstat-no" title="statement not covered" >rom 'data/Media';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span>
&nbsp;
export const feature<span class="cstat-no" title="statement not covered" >dMedia = () =&gt; ({<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span>
  type: MediaType,
  description: 'The featured media for the object.',
  resolve: ({ featured<span class="fstat-no" title="function not covered" >_media: </span>featured }) =&gt; (featur<span class="cstat-no" title="statement not covered" >ed &gt; 0 ? Media.load(<span class="cstat-no" title="statement not covered" ></span>featured) : null),</span>
});
&nbsp;
export const descr<span class="cstat-no" title="statement not covered" >iption = {</span>
  description: {
    type: Description,
  },
};
&nbsp;
export const c<span class="cstat-no" title="statement not covered" >aption = {</span>
  caption: {
    type: Caption,
  },
};
&nbsp;
export const a<span class="cstat-no" title="statement not covered" >ltText = {</span>
  alt_text: {
    type: GraphQLString,
    description: 'Alternative text to display when attachment is not displayed.',
  },
};
&nbsp;
export const med<span class="cstat-no" title="statement not covered" >iaType = {</span>
  media_type: {
    type: MEDIA_TYPE,
  },
};
&nbsp;
export const mi<span class="cstat-no" title="statement not covered" >meType = {</span>
  mime_type: {
    type: GraphQLString,
    description: 'The attachment MIME type.',
  },
};
&nbsp;
export const sou<span class="cstat-no" title="statement not covered" >rceUrl = {</span>
  source_url: {
    type: GraphQLString,
    description: 'URL to the original attachment file.',
  },
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Jun 30 2017 19:51:09 GMT-0400 (EDT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
