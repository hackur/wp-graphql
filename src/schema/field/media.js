import { GraphQLString } from 'graphql';
import MEDIA_TYPE from 'enum/MediaType';
import Description from 'type/Description';
import Caption from 'type/Caption';
import MediaType from 'type/Media';

export const featuredMedia = () => ({
  type: MediaType,
  description: 'The featured media for the object.',
  resolve: (
    { featured_media: featured },
    args,
    context,
    { rootValue: { loaders: { Media } } }
  ) => (featured > 0 ? Media.load(featured) : null),
});

export const description = {
  description: {
    type: Description,
  },
};

export const caption = {
  caption: {
    type: Caption,
  },
};

export const altText = {
  altText: {
    type: GraphQLString,
    description: 'Alternative text to display when attachment is not displayed.',
  },
};

export const mediaType = {
  mediaType: {
    type: MEDIA_TYPE,
  },
};

export const mimeType = {
  mimeType: {
    type: GraphQLString,
    description: 'The attachment MIME type.',
  },
};

export const sourceUrl = {
  sourceUrl: {
    type: GraphQLString,
    description: 'URL to the original attachment file.',
  },
};
