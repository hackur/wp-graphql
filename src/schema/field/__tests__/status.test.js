import { commentStatus, pingStatus } from 'field/status';

describe('Test schema type field definition', () => {
  test('Test comment_status field', () => {
    expect(commentStatus.comment_status.type.name).toMatchSnapshot();
  });

  test('Test ping_status field', () => {
    expect(pingStatus.ping_status.type.name).toMatchSnapshot();
  });
});
