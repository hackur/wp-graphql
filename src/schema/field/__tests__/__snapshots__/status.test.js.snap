// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test schema type field definition Test comment_status field 1`] = `"COMMENT_STATUS"`;

exports[`Test schema type field definition Test ping_status field 1`] = `"PING_STATUS"`;
